build_resolvers/lib/src/build_asset_uri_resolver.dart

```diff
@@ -27,54 +39,47 @@ class BuildAssetUriResolver extends UriResolver {
   /// later). If this happens we don't want to hide the asset from the analyzer.
   final seenAssets = Set<AssetId>();
 
-  Future<void> performResolve(
-      BuildStep buildStep, List<AssetId> entryPoints, AnalysisDriver driver) {
-    // Basic approach is to start at the first file, update it's contents
-    // and see if it changed, then walk all files accessed by it.
-    var visited = Set<AssetId>();
-    var visiting = _FutureGroup();
-
-    void processAsset(AssetId assetId) {
-      visited.add(assetId);
-
-      visiting.add(buildStep.readAsString(assetId).then((contents) {
-        if (_cachedAssetContents[assetId] != contents) {
-          if (_cachedAssetContents.containsKey(assetId)) {
-            var path = assetPath(assetId);
-            resourceProvider.updateFile(path, contents);
-            driver.changeFile(path);
-          } else {
-            resourceProvider.newFile(assetPath(assetId), contents);
-          }
-          _cachedAssetContents[assetId] = contents;
-          var unit = parseDirectives(contents, suppressErrors: true);
-          var dependencies = unit.directives
-              .whereType<UriBasedDirective>()
-              .where((directive) {
-                var uri = Uri.parse(directive.uri.stringValue);
-                return !_ignoredSchemes.any(uri.isScheme);
-              })
-              .map((d) => AssetId.resolve(d.uri.stringValue, from: assetId))
-              .where((id) => id != null)
-              .toSet();
-          _cachedAssetDependencies[assetId] = dependencies;
+  Future<void> performResolve(BuildStep buildStep, List<AssetId> entryPoints,
+      AnalysisDriver driver) async {
+    final states =
+        await crawlAsync<AssetId, _AssetState>(entryPoints, (id) async {
+      try {
+        return _AssetState(id, await buildStep.readAsString(id));
+      } catch (e) {
```

Comment:
We should probably only catch AssetNotFoundException here

Reply:
This was translating what we had before which caught anything, I agree that it would be better to be more specific.